### PR TITLE
refactor: switch from Axios for Fetch API

### PR DESCRIPTION
**Pull request by gauthier-th**
#### Description

This PR switch our codebase from the Axios library to the native Fetch API instead.

#### To-Dos

- [ ] Successful build `yarn build`
- [ ] Translation keys `yarn i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #XXXX

### CODE REVIEW

File: src/components/Layout/UserWarnings/index.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ const UserWarnings: React.FC<UserWarningsProps> = ({ onClick }) => {
   let res = null;
 
   //check if a user has warnings
+  console.log(user);
```
Comment (fallenbagel) [inline src/components/Layout/UserWarnings/index.tsx]: Remove the console log

**Comment by Gauvino** [comment]
Multiple encounter:

First: in movies and series when changing of sort type it break the image render
chrome_r8YWmAOJt2
Second: when adding a filter half of the show or render
chrome_08B0RlEUQO
Third: can't comment on issue
CCUmuYQfZf
Fourth: can't update a user in every section, only permissions one seem's to work, and for editing our account only notifications don't work
chrome_RZkHtQrdBc
Fifth: when clicking on a movie, and clicking on a similar title it render with the image of the previous movie, also similar title movie don't render at all
chrome_Hrw1yTrlBJ
uLTVTPRNXS

**Comment by Fallenbagel** [comment]
@Gauvino your fourth bug, is not a bug. You can't save because of our current email validation. That user needs a proper email in their email field then you can save

**Comment by Fallenbagel** [comment]
We need help on this PR.

Running into an issue where every now and then on not that slow but somewhat slow Internet like 30mbps etc. This might be related to the bugs that @Gauvino posted on their comment.

Fetch request keeps timing out.
```
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async JellyfinAPI.get (/home/fallenbagel/code/jellyseerr/server/api/externalapi.ts:76:24)
    at async JellyfinAPI.getItemData (/home/fallenbagel/code/jellyseerr/server/api/jellyfin.ts:349:28)
    at async JellyfinScanner.processMovie (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:63:24)
    at async /home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:514:11
    at async Promise.all (index 18)
    at async JellyfinScanner.processItems (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:511:5)
    at async JellyfinScanner.loop (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:543:7)
    at async JellyfinScanner.run (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:657:11) {
  cause: ConnectTimeoutError: Connect Timeout Error
      at onConnectTimeout (node:internal/deps/undici/undici:7760:28)
      at node:internal/deps/undici/undici:7716:50
      at Immediate._onImmediate (node:internal/deps/undici/undici:7748:13)
      at processImmediate (node:internal/timers:478:21) {
    code: 'UND_ERR_CONNECT_TIMEOUT'
  }
}
Request took 10004ms
```

So we need input from someone who has any experience in solving this issue

**Comment by M0NsTeRRR** [comment]
> We need help on this PR.
> 
> Running into an issue where every now and then on not that slow but somewhat slow Internet like 30mbps etc. This might be related to the bugs that @Gauvino posted on their comment.
> 
> Fetch request keeps timing out.
> 
> ```
> TypeError: fetch failed
>     at node:internal/deps/undici/undici:12618:11
>     at processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async JellyfinAPI.get (/home/fallenbagel/code/jellyseerr/server/api/externalapi.ts:76:24)
>     at async JellyfinAPI.getItemData (/home/fallenbagel/code/jellyseerr/server/api/jellyfin.ts:349:28)
>     at async JellyfinScanner.processMovie (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:63:24)
>     at async /home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:514:11
>     at async Promise.all (index 18)
>     at async JellyfinScanner.processItems (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:511:5)
>     at async JellyfinScanner.loop (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:543:7)
>     at async JellyfinScanner.run (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:657:11) {
>   cause: ConnectTimeoutError: Connect Timeout Error
>       at onConnectTimeout (node:internal/deps/undici/undici:7760:28)
>       at node:internal/deps/undici/undici:7716:50
>       at Immediate._onImmediate (node:internal/deps/undici/undici:7748:13)
>       at processImmediate (node:internal/timers:478:21) {
>     code: 'UND_ERR_CONNECT_TIMEOUT'
>   }
> }
> Request took 10004ms
> ```
> 
> So we need input from someone who has any experience in solving this issue

Hello,

https://undici.nodejs.org/#/?id=network-address-family-autoselection
`UND_ERR_CONNECT_TIMEOUT` is an issue related to IPv6, not a slow connection according to the documentation. Additionally, `30mbps` is not a slow connection. A timeout of `10004ms` is expected to timeout. To test with a bad connection, open the network tab in your browser and set the speed limitation to 2G (40 kbit/s).

**Comment by gauthier-th** [comment]
> > We need help on this PR.
> > 
> > Running into an issue where every now and then on not that slow but somewhat slow Internet like 30mbps etc. This might be related to the bugs that @Gauvino posted on their comment.
> > 
> > Fetch request keeps timing out.
> > 
> > ```
> > TypeError: fetch failed
> >     at node:internal/deps/undici/undici:12618:11
> >     at processTicksAndRejections (node:internal/process/task_queues:95:5)
> >     at async JellyfinAPI.get (/home/fallenbagel/code/jellyseerr/server/api/externalapi.ts:76:24)
> >     at async JellyfinAPI.getItemData (/home/fallenbagel/code/jellyseerr/server/api/jellyfin.ts:349:28)
> >     at async JellyfinScanner.processMovie (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:63:24)
> >     at async /home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:514:11
> >     at async Promise.all (index 18)
> >     at async JellyfinScanner.processItems (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:511:5)
> >     at async JellyfinScanner.loop (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:543:7)
> >     at async JellyfinScanner.run (/home/fallenbagel/code/jellyseerr/server/lib/scanners/jellyfin/index.ts:657:11) {
> >   cause: ConnectTimeoutError: Connect Timeout Error
> >       at onConnectTimeout (node:internal/deps/undici/undici:7760:28)
> >       at node:internal/deps/undici/undici:7716:50
> >       at Immediate._onImmediate (node:internal/deps/undici/undici:7748:13)
> >       at processImmediate (node:internal/timers:478:21) {
> >     code: 'UND_ERR_CONNECT_TIMEOUT'
> >   }
> > }
> > Request took 10004ms
> > ```
> > 
> > So we need input from someone who has any experience in solving this issue
> 
> Hello,
> 
> https://undici.nodejs.org/#/?id=network-address-family-autoselection
> `UND_ERR_CONNECT_TIMEOUT` is an issue related to IPv6, not a slow connection according to the documentation. Additionally, `30mbps` is not a slow connection. A timeout of `10004ms` is expected to timeout. To test with a bad connection, open the network tab in your browser and set the speed limitation to 2G (40 kbit/s).

Thanks for your response.
These issues are happening on the backend of the application, so it's not possible to debug with browser tools.

For IPv6, we tried things like adding `--dns-result-order=ipv4first` as argument.
I didn't manage to replicate this issue, but @Fallenbagel did so he may have other insights.

We'd be happy to try out other ideas.

**Comment by M0NsTeRRR** [comment]
Thanks for your feedback. `dns-result-order` should not help at all, as it only manages the DNS response order (IPv4 first doesn't prevent you from still getting both IPv4 and IPv6 results).

From the [documentation](https://nodejs.org/api/net.html#netgetdefaultautoselectfamily) if i'm not misleading the default configuration set `getdefaultautoselectfamily` to true which use first `AAAA` record first (if returned), first A, second AAAA, second A etc.

To confirm the IPv6 issue I see two possibilities (I'm not a Node.js expert, so I might miss some steps.) : 
 - Force IPv6 connection with `family` in the TCP socket.
 - Using `NODE_OPTIONS="--dns-result-order=ipv4first --no-network-family-autoselection"` which should return IPv4 first and use the first returned record.

Concerning the root cause, I suspect that Node.js detects your host supports IPv6, but it is misconfigured, which leads to this issue.

**Comment by Gauvino** [comment]
Sorry for the news but nothing as change, all of the bugs are still there!

Does I can provide some logs are something to investigate more?

**Review by Fallenbagel — APPROVED** [review]
LGTM!

**Comment by fallenbagel** [comment]
@all-contributors please add @gauthier-th for maintenance

**Comment by fallenbagel** [comment]
@all-contributors please add @gauthier-th for maintenance

**Pull request merged**